src/api/utils/project.py

```diff
@@ -0,0 +1,165 @@
+    summary = {}
+    protocols = []
+    sample_unit_methods = TransectMethod.__subclasses__()
+    for sample_unit_method in sample_unit_methods:
+        sample_unit_method_summary = _create_submitted_sample_unit_method_summary(
+            sample_unit_method, project
+        )
+
+        if not sample_unit_method_summary:
+            continue
+
+        protocols.append(sample_unit_method.protocol)
+        for site_id in sample_unit_method_summary:
+            if site_id not in summary:
+                summary[site_id] = sample_unit_method_summary[site_id]
+            else:
+                summary[site_id]["sample_unit_methods"].update(
+                    sample_unit_method_summary[site_id]["sample_unit_methods"]
+                )
+
+    return protocols, summary
+
+
+def _get_collect_record_label(collect_record):
+    number_label = []
+    sample_unit = collect_record.sample_unit
+    number = sample_unit.get("number")
+    label = sample_unit.get("label")
+    if isinstance(number, (str, int)):
+        number_label.append(str(number))
+
+    if label:
+        number_label.append(label)
+
+    return "-".join(number_label)
+
+
+def create_collecting_summary(project):
+    DEFAULT_SITE_ID = "__null__"
+    summary = {}
+    collect_records = CollectRecord.objects.select_related("profile").filter(
+        project=project
+    )
+    site_ids = [
+        get_value(collect_record.data or {}, "sample_event.site", ".")
+        for collect_record in collect_records
+    ]
+    sites = Site.objects.filter(id__in=set(site_ids))
+    site_lookup = {str(s.pk): s.name for s in sites}
+
+    protocols = {}
+
+    for collect_record in collect_records:
+        data = collect_record.data or {}
+        protocol = get_value(data, "protocol")
+
+        if protocol not in PROTOCOL_MAP:
+            continue
+
+        protocols[protocol] = None
+
+        profile = collect_record.profile
+        site_id = get_value(data, "sample_event.site", ".")
+        profile_id = str(profile.pk)
+
+        if site_id not in summary:
+            summary[site_id] = {
+                "site_name": site_lookup.get(site_id) or DEFAULT_SITE_ID,
+                "sample_unit_methods": {protocol: {"profile_summary": {}}},
+            }
+
+        if protocol not in summary[site_id]["sample_unit_methods"]:
+            summary[site_id]["sample_unit_methods"][protocol] = {"profile_summary": {}}
+
+        if (
+            profile_id
+            not in summary[site_id]["sample_unit_methods"][protocol]["profile_summary"]
+        ):
+            summary[site_id]["sample_unit_methods"][protocol]["profile_summary"][
+                profile_id
+            ] = {"profile_name": profile.full_name, "labels": []}
+
+        label = _get_collect_record_label(collect_record)
+        summary[site_id]["sample_unit_methods"][protocol]["profile_summary"][
+            profile_id
+        ]["labels"].append(label)
+
+    return list(protocols), summary
```

Comment:
I wanted a unique list, with a list it would look something like below.

```
protocols = []
...
protocols.append(....)
...
protocols.append(....)
...

return list(set(protocols)), summary
```